canonical-web-and-design/maas-ui
# feat(ui): add create tag form

## Done

- Add a form to create tags to the tag list header.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the tag list (under machines).
- Click "Create new tag" in the header.
- Fill out the fields, entering a bad xpath e.g. `//node[@class="system"]/vendor = "QEMU" and,` and submit.
- You should get an error under the xpath field.
- Enter a valid xpath e.g. 
```
//node[@class="system"]/vendor = "QEMU" and
//node[@class="processor"]/vendor[starts-with(.,"Advanced Micro Devices")] and not
//node[@id="firmware"]/capabilities/capability/@id = "uefi"
```
- Submit and you should get redirected to the newly created tag.

## Fixes

Fixes: canonical-web-and-design/app-tribe#747.

## Screenshots
<img width="1220" alt="Screen Shot 2022-03-11 at 3 39 21 pm" src="https://user-images.githubusercontent.com/361637/157803364-8bd3e6c3-55da-410a-af92-278bd191b97f.png">
